plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/ApplicationWorkbenchWindowAdvisor.java

```diff
@@ -468,6 +472,48 @@ private void closeEmptyEditors() {
         }
     }
 
+    private void handlePartClosed(@NotNull IWorkbenchPartReference ref) {
+        if (!DBWorkbench.getPlatform().getPreferenceStore().getBoolean(DatabaseEditorPreferences.PROP_DISCONNECT_ON_EDITORS_CLOSE)) {
+            return;
+        }
+        if (!(ref instanceof IEditorReference editor)) {
+            // Not an editor
+            return;
+        }
+        DBPDataSourceContainer container;
+        try {
+            container = EditorUtils.getInputDataSource(editor.getEditorInput());
+        } catch (PartInitException ignored) {
+            container = null;
+        }
+        if (container != null && !hasEditorsForDataSource(container)) {
+            log.debug("Last editor for '" + container.getName() + "' was closed. Closing connection");
+            DataSourceHandler.disconnectDataSource(container, null);
+        }
+    }
+
+    private boolean hasEditorsForDataSource(@NotNull DBPDataSourceContainer container) {
+        for (IWorkbenchWindow window : PlatformUI.getWorkbench().getWorkbenchWindows()) {
+            for (IWorkbenchPage page : window.getPages()) {
+                for (IEditorReference ref : page.getEditorReferences()) {
+                    if (isEditorForDataSource(ref, container)) {
+                        return true;
+                    }
+                }
+            }
+        }
+
+        return false;
+    }
+
+    private boolean isEditorForDataSource(@NotNull IEditorReference ref, @NotNull DBPDataSourceContainer container) {
+        try {
```

Comment:
It checks all editors including non-activated.
Is it expected?

Reply:
Yes - otherwise, it might result in unwanted disconnects.

Let's say you have three tabs:
```
[ Album (deactivated) ] [ Artist (deactivated) ] [ Invoice (activated, selected) ]
```
And you want to close the "Invoice" tab. Since it's the only activated editor, the connection can be closed. After that, the "Artist" tab is activated, and since the connection is closed, it must be opened again. Hence, this method considers ALL editors, whether they're activated or not

Reply:
Then this feature doesn't make much sense. Quite often there are multiple SQL editors open and most of them are not active. In this case connection will never close.

Reply:
The idea was to close a connection when there are no more editors left regarding whether they're active or not - this makes sense in my head with an explanation in the form of an example above.

Reply:
We can schedule an overview of this feature in the following UX discussion next week